### ISSUE TITLE
CCL couldn't get local_idx/count from environment variables on XPU 

### ISSUE THREAD

**Issue by faaany** [huggingface/accelerate#2340]
### System Info

```Shell
- `Accelerate` version: 0.27.0.dev0
- Platform: Linux-5.15.0-91-generic-x86_64-with-glibc2.35
- Python version: 3.10.13
- Numpy version: 1.26.3
- PyTorch version (GPU?): 2.1.0a0+cxx11.abi (False)
- PyTorch XPU available: True
- PyTorch NPU available: False
- System RAM: 1007.61 GB
- `Accelerate` default config:
        Not found
```

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [X] My own task or dataset (give details below)

### Reproduction
My code is based on this [script](https://github.com/tloen/alpaca-lora/blob/main/finetune.py). And the command to launch my job is `accelerate launch training.py`. 

```bash 
2024:01:15-03:36:29:(1980058) |CCL_WARN| did not find MPI-launcher specific variables, switch to ATL/OFI, to force enable ATL/MPI set CCL_ATL_TRANSPORT=mpi
2024:01:15-03:36:29:(1980058) |CCL_WARN| could not get local_idx/count from environment variables, trying to get them from ATL
2024:01:15-03:36:29:(1980058) |CCL_ERROR| global.cpp:150 getenv_local_coord: condition global_data::env().ze_ipc_exchange == ccl::ze::ipc_exchange_mode::sockets failed
to get local_idx/count from ATL, set CCL_ZE_IPC_EXCHANGE=sockets explicitly
Traceback (most recent call last):
  File "/soft/fanli/HuggingFace/testcase/fine-tune/run_fine-tune.py", line 261, in <module>
    fire.Fire(train)
  File "/home/sdp/miniconda3/envs/study/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/home/sdp/miniconda3/envs/study/lib/python3.10/site-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/home/sdp/miniconda3/envs/study/lib/python3.10/site-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/soft/fanli/HuggingFace/testcase/fine-tune/run_fine-tune.py", line 254, in train
    trainer.train()
  File "/soft/fanli/hardware-study/transformers/src/transformers/trainer.py", line 1539, in train
    return inner_training_loop(
  File "/soft/fanli/hardware-study/transformers/src/transformers/trainer.py", line 1687, in _inner_training_loop
    model, self.optimizer = self.accelerator.prepare(self.model, self.optimizer)
  File "/soft/fanli/hardware-study/accelerate/src/accelerate/accelerator.py", line 1227, in prepare
    result = tuple(
  File "/soft/fanli/hardware-study/accelerate/src/accelerate/accelerator.py", line 1228, in <genexpr>
    self._prepare_one(obj, first_pass=True, device_placement=d) for obj, d in zip(args, device_placement)
  File "/soft/fanli/hardware-study/accelerate/src/accelerate/accelerator.py", line 1104, in _prepare_one
    return self.prepare_model(obj, device_placement=device_placement)
  File "/soft/fanli/hardware-study/accelerate/src/accelerate/accelerator.py", line 1355, in prepare_model
    model = torch.nn.parallel.DistributedDataParallel(
  File "/home/sdp/miniconda3/envs/study/lib/python3.10/site-packages/torch/nn/parallel/distributed.py", line 795, in __init__
    _verify_param_shape_across_processes(self.process_group, parameters)
  File "/home/sdp/miniconda3/envs/study/lib/python3.10/site-packages/torch/distributed/utils.py", line 265, in _verify_param_shape_across_processes
    return dist._verify_params_across_processes(process_group, tensors, logger)
RuntimeError: oneCCL: global.cpp:150 getenv_local_coord: EXCEPTION: to get local_idx/count from ATL, set CCL_ZE_IPC_EXCHANGE=sockets explicitly
2024:01:15-03:36:31:(1980057) |CCL_WARN| did not find MPI-launcher specific variables, switch to ATL/OFI, to force enable ATL/MPI set CCL_ATL_TRANSPORT=mpi
2024:01:15-03:36:31:(1980057) |CCL_WARN| could not get local_idx/count from environment variables, trying to get them from ATL
2024:01:15-03:36:31:(1980057) |CCL_ERROR| global.cpp:150 getenv_local_coord: condition global_data::env().ze_ipc_exchange == ccl::ze::ipc_exchange_mode::sockets failed
to get local_idx/count from ATL, set CCL_ZE_IPC_EXCHANGE=sockets explicitly
Traceback (most recent call last):
  File "/soft/fanli/HuggingFace/testcase/fine-tune/run_fine-tune.py", line 261, in <module>
    fire.Fire(train)
  File "/home/sdp/miniconda3/envs/study/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/home/sdp/miniconda3/envs/study/lib/python3.10/site-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/home/sdp/miniconda3/envs/study/lib/python3.10/site-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/soft/fanli/HuggingFace/testcase/fine-tune/run_fine-tune.py", line 254, in train
    trainer.train()
  File "/soft/fanli/hardware-study/transformers/src/transformers/trainer.py", line 1539, in train
    return inner_training_loop(
  File "/soft/fanli/hardware-study/transformers/src/transformers/trainer.py", line 1687, in _inner_training_loop
    model, self.optimizer = self.accelerator.prepare(self.model, self.optimizer)
  File "/soft/fanli/hardware-study/accelerate/src/accelerate/accelerator.py", line 1227, in prepare
    result = tuple(
  File "/soft/fanli/hardware-study/accelerate/src/accelerate/accelerator.py", line 1228, in <genexpr>
    self._prepare_one(obj, first_pass=True, device_placement=d) for obj, d in zip(args, device_placement)
  File "/soft/fanli/hardware-study/accelerate/src/accelerate/accelerator.py", line 1104, in _prepare_one
    return self.prepare_model(obj, device_placement=device_placement)
  File "/soft/fanli/hardware-study/accelerate/src/accelerate/accelerator.py", line 1355, in prepare_model
    model = torch.nn.parallel.DistributedDataParallel(
  File "/home/sdp/miniconda3/envs/study/lib/python3.10/site-packages/torch/nn/parallel/distributed.py", line 795, in __init__
    _verify_param_shape_across_processes(self.process_group, parameters)
  File "/home/sdp/miniconda3/envs/study/lib/python3.10/site-packages/torch/distributed/utils.py", line 265, in _verify_param_shape_across_processes
    return dist._verify_params_across_processes(process_group, tensors, logger)
RuntimeError: oneCCL: global.cpp:150 getenv_local_coord: EXCEPTION: to get local_idx/count from ATL, set CCL_ZE_IPC_EXCHANGE=sockets explicitly
[2024-01-15 03:36:35,030] torch.distributed.elastic.multiprocessing.api: [ERROR] failed (exitcode: 1) local_rank: 0 (pid: 1980057) of binary: /home/sdp/miniconda3/envs/study/bin/python
Traceback (most recent call last):
  File "/home/sdp/miniconda3/envs/study/bin/accelerate", line 8, in <module>
    sys.exit(main())
  File "/soft/fanli/hardware-study/accelerate/src/accelerate/commands/accelerate_cli.py", line 47, in main
    args.func(args)
  File "/soft/fanli/hardware-study/accelerate/src/accelerate/commands/launch.py", line 1014, in launch_command
    multi_gpu_launcher(args)
  File "/soft/fanli/hardware-study/accelerate/src/accelerate/commands/launch.py", line 672, in multi_gpu_launcher
    distrib_run.run(args)
  File "/home/sdp/miniconda3/envs/study/lib/python3.10/site-packages/torch/distributed/run.py", line 797, in run
    elastic_launch(
  File "/home/sdp/miniconda3/envs/study/lib/python3.10/site-packages/torch/distributed/launcher/api.py", line 134, in __call__
    return launch_agent(self._config, self._entrypoint, list(args))
  File "/home/sdp/miniconda3/envs/study/lib/python3.10/site-packages/torch/distributed/launcher/api.py", line 264, in launch_agent
    raise ChildFailedError(
torch.distributed.elastic.multiprocessing.errors.ChildFailedError:
============================================================
fine-tune/run_fine-tune.py FAILED
------------------------------------------------------------
Failures:
[1]:
  time      : 2024-01-15_03:36:35
  host      : a4bf01943df8.jf.intel.com
  rank      : 1 (local_rank: 1)
  exitcode  : 1 (pid: 1980058)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
------------------------------------------------------------
Root Cause (first observed failure):
[0]:
  time      : 2024-01-15_03:36:35
  host      : a4bf01943df8.jf.intel.com
  rank      : 0 (local_rank: 0)
  exitcode  : 1 (pid: 1980057)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
============================================================
```
### Root-Cause
The error log above say that oneCCL could not get local rank from environment variables and [here](https://github.com/oneapi-src/oneCCL/blob/8d18c7b5cc1669a1ae4cc51eeaedbfbada8c1f46/src/common/global/global.cpp#L200) is the oneCCL source code. The reason of the error is that I am not using `mpirun -f hosts -n 2 -ppn 2 python xx.py`, but `accelerate launch xx.py` to launch my distributed job. And by default oneCCL recognizes MPI-Launcher. 

In order to make it work, we need to explicitly set the environment `CCL_PROCESS_LAUNCHER` as specified [here](https://oneapi-src.github.io/oneCCL/env-variables.html#ccl-process-launcher). Among the 3 options, we should go for "none" and explicitly set the `CCL_LOCAL_SIZE` and `CCL_LOCAL_RANK`. Then both `mpirun` and `accelerate launch` will work for distributed training on XPU.

### Expected behavior
I should not get this error when running distributed fine-tuning on XPU and somewhere in accelerate should set this environment variables for me.

**Comment by faaany** [comment]
Here is the PR for this issue: https://github.com/huggingface/accelerate/pull/2339. @muellerzr pls have a review. Thanks!

**Comment by BenjaminBossan** [comment]
Resolved via #2339.
